src/Language/PureScript/Parser/Lexer.hs

```diff
@@ -173,6 +174,12 @@ updatePositions (x:xs) = x : zipWith update (x:xs) xs
 parseTokens :: Lexer u [PositionedToken]
 parseTokens = whitespace *> P.many parsePositionedToken <* P.skipMany parseComment <* P.eof
 
+lexLenient :: FilePath -> Text -> Either P.ParseError [PositionedToken]
+lexLenient f s = updatePositions <$> P.parse parseTokensLenient f s
+
+parseTokensLenient :: Lexer u [PositionedToken]
```

Comment:
Where should I put unit tests for compiler code? I can add them to purs ide's suite, which kind of steals the usual namespace for hspec test suites.